jobs: add button to request execution details

adityamaru opened the pull request:
This is the last of the three PRs to add support
for requesting, viewing and downloading execution
details from the job details page.

This change wires up the logic needed to request
the execution details for a given job. The request
is powered by the crdb_internal.request_job_execution_details
builtin that triggers the collection of execution details.

Fixes: https://github.com/cockroachdb/cockroach/issues/105076
Release note: None

adityamaru committed:
jobs: add button to request execution details

This is the last of the three PRs to add support
for requesting, viewing and downloading execution
details from the job details page.

This change wires up the logic needed to request
the execution details for a given job. The request
is powered by the crdb_internal.request_job_execution_details
builtin that triggers the collection of execution details.

This change also adds logic to hide the Advanced Debugging
tab if the user is not an admin.

Fixes: #105076
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107759)
<!-- Reviewable:end -->

adityamaru commented:
Loom video: https://www.loom.com/share/57ec04141b1443c2b41a630e1771dd2c?sid=0610ee20-cb72-4952-89ae-8c1a2202e2f2

Fist commit is from https://github.com/cockroachdb/cockroach/pull/107210 which should be merged soon.

maryliag reviewed (commented):
Discussed offline, but adding here for tracking:
- Hide the Advanced Debugging tab if the user doesn't have ADMIN or VIEWACTIVITY role
- Change the request button to keep consistent with other pages, so move it to the right and increase its height, an example that can be follow is the Activate Diagnostics on the Statement Details page

![Screenshot 2023-07-28 at 9.53.02 AM.png](https://files.reviewableusercontent.io/images/reviewable/1017486/lFM8g2wV4C-PtglMohED5qfm/Screenshot%202023-07-28%20at%209.53.02%20AM.png)

> Reviewed 12 of 12 files at r1, 2 of 10 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107759#-:-NaRla1m22lVXni_yWa-:bry81u6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt)



<!-- Sent from Reviewable.io -->

adityamaru commented:
thanks for your help! I changed the appearance of the button and also added logic to only show the tab if we are admin. This may change in the future but for now I think its better to take a conservative approach.

<img width="1537" alt="Screenshot 2023-07-28 at 3 22 40 PM" src="https://github.com/cockroachdb/cockroach/assets/13837382/b64dd03d-5e74-4e0c-bbc3-6390e242e7f3">

maryliag reviewed (approved):
Nice!
The only thing that I would like to see is now with the permission checks, the page loading for an admin and then how it shows for non-admins, just to confirm everything is working as expected.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 6 of 10 files at r2, 9 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107759#-:-Namwzp5BRBFZf14O4HA:byxon1l)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @adityamaru and @dt)

___
*[`pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.tsx` line 118 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107759#-NaSLs112ioBAUVH4w4R:-NaSLs112ioBAUVH4w4S:b-wuscsc) ([raw file](https://github.com/cockroachdb/cockroach/blob/f23c46835f39510c7395b86b08e50ad75da37c42/pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.tsx#L118)):*
> ```XML
>                 onDownloadExecutionFileClicked(req).then(resp => {
>                   const type = getContentTypeForFile(executionDetailFile);
>                   const executionFileBytes = new Blob([resp.data], {
> ```

add `resp?.data` because when initially load you might not have the resp yet and this can cause a crash


<!-- Sent from Reviewable.io -->

adityamaru commented:
DBConsole as a non-admin user:

<img width="1731" alt="Screenshot 2023-08-02 at 11 58 45 AM" src="https://github.com/cockroachdb/cockroach/assets/13837382/2d4e5897-a732-442c-8f35-ecdc0361eba6">

Granted the user admin and refreshed the page:

<img width="1728" alt="Screenshot 2023-08-02 at 11 59 08 AM" src="https://github.com/cockroachdb/cockroach/assets/13837382/d7b347c7-e96d-4d18-9e94-127287c6afb3">

adityamaru reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107759)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt and @maryliag)

___
*[`pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.tsx` line 118 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107759#-NaSLs112ioBAUVH4w4R:-NaqyfN49ElATlM9T-O1:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f23c46835f39510c7395b86b08e50ad75da37c42/pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.tsx#L118)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

add `resp?.data` because when initially load you might not have the resp yet and this can cause a crash
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityamaru commented:
TFTR!

bors r=maryliag
